websites/D/Duolingo/presence.ts

```diff
@@ -73,91 +80,107 @@ const presence = new Presence({
 		freezes: null as number,
 		lessonTimeStamp: null as number,
 		inLesson: false,
+		finishedLesson: null as string,
 	},
-	users: { username: string; displayName: string; img: string }[] = [];
+	storedUsers = localStorage.getItem("PMD-users-cache"),
+	users: { username: string; displayName: string; img: string }[] = storedUsers
+		? JSON.parse(storedUsers)
+		: [];
 
 let timeoutId: number;
 
 function newTimeStamp() {
 	return Math.floor(Date.now() / 1000);
 }
 
-function makePossessive(name: string) {
-	return name.endsWith("s") ? `${name}'` : `${name}'s`;
-}
-
-function deEsser(word: string) {
-	return word.endsWith("s") ? word.slice(0, -1) : word;
-}
-
-function giveArticle(word: string) {
-	return `${
-		["a", "e", "i", "o", "u"].some(vowel =>
-			word.toLowerCase().startsWith(vowel)
-		)
-			? "an"
-			: "a"
-	} ${word}`;
-}
-
-function makeProgressBar(value: number, maxValue: number, size: number) {
-	const progressPercentage = Math.min(
-			100,
-			Math.max(0, Math.floor((value / maxValue) * 100))
-		),
-		completedSquares = Math.floor((progressPercentage * size) / 100);
-	return `${"🟩".repeat(completedSquares)}${"⬛".repeat(
-		size - completedSquares
-	)} ${progressPercentage}%`;
-}
-
-function handleLesson() {
-	user.inLesson = true;
+function handleLesson(_path: string | string[]) {
+	const progressBarElement =
+			document.querySelector('div[role="progressbar"]') ?? null,
```

Comment:
`?? null` is somewhat redundant
```suggestion
			document.querySelector('div[role="progressbar"]'),
```

Reply:
No no, I remember it crying about a null check

Reply:
Will check when I'm home

Reply:
Had to put it in for something I scrapped, forgot to remove it